src/main/java/games/strategy/engine/lobby/server/db/HashedPassword.java

```diff
@@ -4,6 +4,10 @@
 
 import games.strategy.util.MD5Crypt;
 
+/**
+ * A Wrapper class for password hashes in the form of the legacy MD5 with salt system,
```

Comment:
We probably should clarify what this class is meant to represent, javadoc is good for that, mea culpa for not having one before hand. To me a 'hashedPassword' is not the raw password provided by the client, but it is the value we received and then hashed before storing or comparing to what we have in DB. A `hashedPassword` is therefore something that is very close to DB, it's basically a value that we could have exactly in database.

Given there is so much hashing going on, I can see it being confusing, this class is not meant to represent the raw password the server gets from the client, whether that value is hashed or not. I can see that being very confusing. With the encryption going on, after server decrypts and removes salt, that would be the raw password from the server point of view, and after bcrypting that value do we then get a `HashedPassword`